packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -320,6 +335,10 @@ - (instancetype)initWithPlayerItem:(AVPlayerItem *)item
   _videoOutput = [avFactory videoOutputWithPixelBufferAttributes:pixBuffAttributes];
   frameUpdater.videoOutput = _videoOutput;
 
+  _pixelBufferSynchronizationQueue =
+      dispatch_queue_create("io.flutter.video_player.pixelBufferSynchronizationQueue", NULL);
```

Comment:
You are blocking the main thread on this queue in copyPixelBuffer, so it should have a higher QoS than the default.

Reply:
Documentation says it is called on "raster" thread.

Reply:
Oops, I didn't realize `copyPixelBuffer` didn't follow the usual engine convention of main thread calling. Good catch!